openpgp/packet/private_key.go

```diff
@@ -482,19 +531,61 @@ func (pk *PrivateKey) encrypt(passphrase []byte, blockCipher CipherFunction, s2k
 	return err
 }
 
-// encryptWithConfig encrypts an unencrypted private key using a passphrase and with the config.
-func (pk *PrivateKey) encryptWithConfig(passphrase []byte, c *Config) error {
-	return pk.encrypt(passphrase, c.Cipher(), c.S2K())
+// EncryptWithConfig encrypts an unencrypted private key using the passphrase and the configuration.
+func (pk *PrivateKey) encryptWithConfig(passphrase []byte, configuration *Config) error {
+	parameters, err := s2k.Generate(configuration.Random(), configuration.S2K())
```

Comment:
Also here since the type is `Params` I would just call this `params` (and in the next function as well)